### PR TITLE
[WIP]Pytest Style analysis/test_align.py

### PR DESCRIPTION
**Pull request by utkbansal**
Fixes #

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by kain88-de** [comment]
build fails

**Comment by kain88-de** [comment]
build still fails

**Comment by utkbansal** [comment]
@kain88-de Build passed without changing anything.

File: testsuite/MDAnalysisTests/analysis/test_align.py [hunk 1]
```diff
@@ -20,153 +20,134 @@
...
+        assert rmsd is None
 
     def test_wrong_dtype(self):
-        rot, rmsd = align.rotation_matrix(self.a.astype(np.int),
-                                          self.b.astype(np.int),
-                                          self.w.astype(np.float32))
+        rot, rmsd = align.rotation_matrix(self.a.astype(np.int), self.b.astype(np.int), self.w.astype(np.float32))
         assert_equal(rot, np.eye(3))
-        assert_equal(rmsd, None)
+        assert rmsd is None
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_align.py]: I have a feeling the first 3 tests can be parametrized.

File: testsuite/MDAnalysisTests/analysis/test_align.py [hunk 2]
```diff
@@ -20,153 +20,134 @@
...
         rmsd = rms.rmsd(first_frame, last_frame, superposition=True)
-        assert_almost_equal(rms.rmsd(last_frame, first_frame,
-                                     superposition=True),
-                            rmsd, 6,
-                            err_msg="error: rmsd() is not symmetric")
-        assert_almost_equal(rmsd, 6.820321761927005, 5,
-                            err_msg="RMSD calculation between 1st and last "
-                            "AdK frame gave wrong answer")
+        assert_almost_equal(rms.rmsd(last_frame, first_frame, superposition=True), rmsd, 6, err_msg="error: rmsd() is not symmetric")
+        assert_almost_equal(rmsd, 6.820321761927005, 5, err_msg="RMSD calculation between 1st and last AdK frame gave wrong answer")
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_align.py]: please avoid these super long lines. The pep8 standard is 80 chars

File: testsuite/MDAnalysisTests/analysis/test_align.py [hunk 3]
```diff
@@ -20,153 +20,134 @@
...
 
-    def test_rmsd_deprecated(self):
-        last_atoms_weight = self.universe.atoms.masses
-        A = self.universe.trajectory[0]
-        B = self.reference.trajectory[-1]
+    def test_rmsd_deprecated(self, universe, reference):
+        last_atoms_weight = universe.atoms.masses
+        A = universe.trajectory[0]
+        B = reference.trajectory[-1]
         with warnings.catch_warnings(record=True) as warn:
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_align.py]: can use the warnings catcher from pytest `pytest.warns`

File: testsuite/MDAnalysisTests/analysis/test_align.py (hunk 1 above)
Comment (utkbansal) [inline testsuite/MDAnalysisTests/analysis/test_align.py]: What do I name it?

Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_align.py]: test_rotation_matrix_input?

Comment (utkbansal) [inline testsuite/MDAnalysisTests/analysis/test_align.py]: done.

File: testsuite/MDAnalysisTests/analysis/test_align.py [hunk 4]
```diff
@@ -20,153 +20,137 @@
...
+    w = np.array([1.3, 2.3])
 
-    def test_no_solution_with_weights(self):
-        rot, rmsd = align.rotation_matrix(self.a, self.b, self.w)
+    @pytest.mark.parametrize('a, b, weights', (
+            (a, b, None),
+            (a, b, w),
+            (a.astype(np.int), b.astype(np.int), w.astype(np.float32))
+    ))
+    def test_rotation_matrix_input(self, a, b, weights):  # TODO: Fix the name
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_align.py]: you left the fixme

**Pull request merged**